packages/esm-service-queues-app/src/hooks/useQueueEntries.ts

```diff
@@ -69,104 +53,29 @@ export function useMutateQueueEntries() {
-  // page zero, waits until data is fetched, then fetches the next page, and so on.
-  //
-  // Fine so far. Where things get complicated is in supporting mutation. When a mutation is made, the
-  // SWR hook first returns stale data with `isValidating` set to false. At this point we say we are
-  // "waiting for mutate," because we have called mutate, but the useSWR hook hasn't updated properly
-  // for it yet. Next it returns stale data again, this time with `isValidating` set to true. At this
-  // point we say we are no longer waiting for mutate. Finally, it returns fresh data with `isValidating`
-  // again set to false. We may then update the data array and move on to the next page.
-  const { mutateQueueEntries } = useMutateQueueEntries();
-
-  const [currentPage, setCurrentPage] = useState(0);
-  const [currentRep, setCurrentRep] = useState(rep);
-  const [currentSearchCriteria, setCurrentSearchCriteria] = useState(searchCriteria);
-  const [data, setData] = useState<Array<Array<QueueEntry>>>([]);
-  const [error, setError] = useState<Error | undefined>(undefined);
-  const [pageUrl, setPageUrl] = useState(getInitialUrl(currentRep, currentSearchCriteria));
-  const [totalCount, setTotalCount] = useState(0);
-  const [waitingForMutate, setWaitingForMutate] = useState(false);
-
-  const refetchAllData = useCallback(
-    (newRep: string = currentRep, newSearchCriteria: QueueEntrySearchCriteria = currentSearchCriteria) => {
-      setWaitingForMutate(true);
-      setCurrentPage(0);
-      setPageUrl(getInitialUrl(newRep, newSearchCriteria));
+  const initialMount = useRef<boolean>(true);
+  const [pageUrl, setPageUrl] = useState<string>(getInitialUrl(rep, searchCriteria));
+  const { data, mutate, ...rest } = useOpenmrsFetchAll<QueueEntry>(pageUrl, {
+    swrInfiniteConfig: {
+      revalidateFirstPage: false,
     },
-    [currentRep, currentSearchCriteria],
-  );
+  });
 
-  // This hook listens to the searchCriteria and rep values and refetches the data when they change.
   useEffect(() => {
-    const isSearchCriteriaUpdated = !isEqual(currentSearchCriteria, searchCriteria);
-    const isRepUpdated = currentRep !== rep;
-    if (isSearchCriteriaUpdated || isRepUpdated) {
-      if (isSearchCriteriaUpdated) {
-        setCurrentSearchCriteria(searchCriteria);
-      }
-      if (isRepUpdated) {
-        setCurrentRep(rep);
-      }
-      refetchAllData(rep, searchCriteria);
-    }
-  }, [currentRep, currentSearchCriteria, refetchAllData, rep, searchCriteria, setCurrentSearchCriteria]);
-
-  const { data: pageData, isValidating, error: pageError } = useSWR<QueueEntryResponse, Error>(pageUrl, openmrsFetch);
+    setPageUrl(getInitialUrl(rep, searchCriteria));
+  }, [searchCriteria, rep]);
 
   useEffect(() => {
-    const nextUrl = getNextUrlFromResponse(pageData);
-    const stillWaitingForMutate = waitingForMutate && !isValidating;
-    if (waitingForMutate && isValidating) {
-      setWaitingForMutate(false);
+    if (initialMount) {
```

Comment:
First off, this `initialMount` is a ref, so it is always true. You want `initialMount.current`.

But moreover, if `pageUrl` changes, then `useOpenmrsFetchAll` will get called with the new value of `pageUrl` as a parameter. Why would you call `mutate` at the same time?